microcosm_flask/conventions/upload.py

```diff
@@ -0,0 +1,44 @@
+"""
+Conventions for file upload.
+
+"""
+from flask import request
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.operations import Operation
+from werkzeug.exceptions import BadRequest
+
+
+class UploadConvention(Convention):
+
+    def configure_upload(self, ns, definition):
+        """
+        Register an upload endpoint.
+
+        The definition's func should be an upload function, which must:
+        - accept kwargs for path data
+        - accept a dict of files
+
+        :param ns: the namespace
+        :param definition: the endpoint definition
+
+        """
+        @self.graph.route(ns.collection_path, Operation.Upload, ns)
+        def upload(**path_data):
+            files = request.files
+
+            if not files:
+                raise BadRequest("No files uploaded!")
+
+            definition.func(files, **path_data)
+            return "", Operation.Upload.value.default_code
```

Comment:
Limitation 2: This is not returning anything. We could support a return value from `func` and an optional schema , but it comes at the cost of making the swagger definition smarter about multiple response types.